File: sprint04/JMS/PGS_42747.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int i(0);
    
    sort(citations.begin(), citations.end());

    for (i = 0; i < citations.size(); i++) {
        if (citations[i] >= citations.size() - i)
            return (citations.size() - i);
    }
}

File: sprint04/KMS/First week/PGS_42747.py
def solution(citations):
    answer = 0
    count = []
    for h in range(max(citations)):
        print(h)
        cnt = 0
        cnt1 = 0
        for i in citations:
            if h <= i:
                cnt += 1
            if h >= i:
                cnt1 += 1
        if cnt >= h and cnt1 <= h and answer < h:
            answer = h
    return answer